### ISSUE TITLE
TF 2.2.0-rc2 tf.Keras API:  model.loss_functions no longer exists

### ISSUE THREAD

**Issue by amaiya** [tensorflow/tensorflow#38269]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow):  `Yes`
- OS Platform and Distribution:  `Ubuntu 18.04`
- TensorFlow installed from (binary): `v2.2.0-rc1-34-ge6e5d6df2a 2.2.0-rc2`
- CUDA/cuDNN version: `cuda-10.1`

**Describe the current behavior**
In `tf.Keras` API in  TensorFlow 2.2.0, `model.loss_functions` no longer exists.  Instead the loss functions are buried and seem to be only accessible with code like this:
```python
model.compiled_loss._get_loss_object(model.compiled_loss._losses).fn
```
Being able to easily access loss functions associated with a model is important for many applications that are built on TensorFlow.  Moreover, this is an undocumented breaking change.

**Describe the expected behavior**
The loss functions associated with a model should be more directly and easily accessible like `model.loss_functions`, which is the way loss functions are accessed in TensorFlow 2.1.0.

**Standalone code to reproduce the issue** 
```python
from tensorflow.keras.models import Sequential  
from tensorflow.keras.layers import Dense, Activation  
model = Sequential()  
model.add(Dense(10, input_dim=10, activation='softmax'))  
model.compile(optimizer='sgd', loss='categorical_crossentropy', metrics=['accuracy']) 

# works in TF 2.2.0-rc2, but does NOT work in previous versions like 2.1.0
model.compiled_loss._get_loss_object(model.compiled_loss._losses).fn

# does NOT work in TF 2.2.0-rc2, but works in TF 2.1.0
model.loss_functions[0].fn

**Comment by mihaimaruseac** [comment]
This looks related to #38150

**Comment by amaiya** [comment]
The `tf.keras.losses` module does not allow  you to identify the loss function tied to a compiled model.  So, this solution isn't really satisfactory.  The loss function associated with a compiled model should be publicly accessible, which was part of the point of this issue.

**Comment by pavithrasv** [comment]
Oh i meant you can use `tf.keras.losses.get` to get the loss function using a string. Since loss value is passed to the compile API it should be easy to compile list of loss functions from what's being passed. You can also use `model.compiled_loss.metrics`.

**Comment by amaiya** [comment]
I see - thanks.   

So, the idea is to obtain the loss function string names and supply that to `tf.keras.losses.get`.   In situations where you may not have access to the loss function string names originally supplied to model.compile (e.g., libraries that accept a compiled model as input to facilitate training), what is the best way to obtain the loss function string names directly from the model?  If a model is compiled with a loss function using a string name or loss object, it would make sense for it be possible to query the model and retrieve the loss function (or loss function string name) from the model using public API.

You suggested `model.compiled_loss.metrics`, but this is an empty list using the example above.  Only  `model.compiled_loss._losses` contains the string names:
```
In [1]: from tensorflow.keras.models import Sequential   
   ...: from tensorflow.keras.layers import Dense, Activation   
   ...: model = Sequential()   
   ...: model.add(Dense(10, input_dim=10, activation='softmax'))   
   ...: model.compile(optimizer='sgd', loss='categorical_crossentropy', metrics=['accuracy'])  

In [2]: model.compiled_loss.metrics                                                                                                                                                                         
Out[2]: []

In [3]: model.compiled_loss._losses                                                                                                                                                                         
Out[3]: 'categorical_crossentropy'
```

**Comment by pavithrasv** [comment]
`model.compiled_loss.metrics` gets populated after a model is actually trained /evaluated.

**Comment by amaiya** [comment]
I see that `model.compiled_loss.metrics` is populated after training, but it doesn't return what you claim it returns:

```
In [18]: model.compiled_loss.metrics # no loss function string name                                                                                                                                                                         
Out[18]: [<tensorflow.python.keras.metrics.Mean at 0x7fc85af8a6d8>]
```
There should be an easy and public way to query a model to identify either the loss function or loss function string name (in this case 'categorical_crossentropy') without resorting to underscore atrributes like this::
```
In [19]: model.compiled_loss._losses                                                                                                                                                                        
Out[19]: [<tensorflow.python.keras.losses.LossFunctionWrapper at 0x7fc85aed4a90>]

In [20]: model.compiled_loss._losses[0].fn                                                                                                                                                                  
Out[20]: <function tensorflow.python.keras.losses.categorical_crossentropy(y_true, y_pred, from_logits=False, label_smoothing=0)>

```
